packages/core/streams/ai-stream.ts

```diff
@@ -5,110 +5,142 @@ import {
-          const data = event.data
-          if (data === '[DONE]') {
+      eventSourceParser = createParser(
+        (event: ParsedEvent | ReconnectInterval) => {
+          if (
+            'data' in event &&
+            event.type === 'event' &&
+            event.data === '[DONE]'
+          ) {
             controller.terminate()
             return
           }
 
-          const message = customParser(data)
-          if (message) controller.enqueue(message)
+          if ('data' in event) {
+            const parsedMessage = customParser(event.data)
+            if (parsedMessage) controller.enqueue(parsedMessage)
+          }
         }
-      }
-
-      parser = createParser(onParse)
+      )
     },
 
     transform(chunk) {
-      parser.feed(decoder.decode(chunk))
+      eventSourceParser.feed(textDecoder.decode(chunk))
     }
   })
 }
 
 /**
- * This stream forks input stream, allowing us to use the result as a
- * bytestream of the messages and pass the messages to our callback interface.
+ * Creates a transform stream that calls back with messages.
+ * @param {AIStreamCallbacks} callbacks - The callbacks.
+ * @return {TransformStream<string, Uint8Array>} A transform stream.
  */
 export function createCallbacksTransformer(
   callbacks: AIStreamCallbacks | undefined
-) {
-  const encoder = new TextEncoder()
-  let fullResponse = ''
-
+): TransformStream<string, Uint8Array> {
+  const textEncoder = new TextEncoder()
+  let aggregatedResponse = ''
   const { onStart, onToken, onCompletion } = callbacks || {}
 
-  return new TransformStream<string, Uint8Array>({
+  return new TransformStream({
     async start(): Promise<void> {
       if (onStart) await onStart()
     },
 
     async transform(message, controller): Promise<void> {
-      controller.enqueue(encoder.encode(message))
+      controller.enqueue(textEncoder.encode(message))
 
       if (onToken) await onToken(message)
-      if (onCompletion) fullResponse += message
+      if (onCompletion) aggregatedResponse += message
     },
 
     async flush(): Promise<void> {
-      await onCompletion?.(fullResponse)
+      if (onCompletion) await onCompletion(aggregatedResponse)
     }
   })
 }
 
-// If we're still at the start of the stream, we want to trim the leading
-// `\n\n`. But, after we've seen some text, we no longer want to trim out
-// whitespace.
-export function trimStartOfStreamHelper() {
-  let start = true
-  return (text: string) => {
-    if (start) text = text.trimStart()
-    if (text) start = false
+/**
+ * Creates a function to trim the start of a stream.
+ * @return {Function} A function that trims the start of a stream.
+ */
+export function trimStartOfStreamHelper(): (text: string) => string {
+  let isStreamStart = true
+
+  return (text: string): string => {
+    if (isStreamStart) {
+      text = text.trimStart()
+      if (text) isStreamStart = false
+    }
     return text
   }
 }
 
+/**
+ * Creates an AIStream from the response.
```

Comment:
Comment needs more detail. Comments should not include the name of the method they are describing in the first sentence. Should be phrased like `Returns a ...`